### PR TITLE
CI: Setup SBT manually

### PR DESCRIPTION
**Pull request by ptomato**
One of the core developers of SBT has kindly created an action to replace the missing SBT in newer Ubuntu images.
https://github.com/marketplace/actions/setup-sbt-installer

Closes: #4267

### CODE REVIEW

File: .github/workflows/esmeta-test262.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ jobs:
         with:
           distribution: temurin
           java-version: 17
+      - name: Setup SBT
+        uses: sbt/setup-sbt@v1
```
Comment (ljharb) [inline .github/workflows/esmeta-test262.yml:18]: can we skip this step when sbt is already present?

Comment (ptomato) [reply .github/workflows/esmeta-test262.yml:18]: I'm not so familiar with GitHub actions that I know how to do that off the top of my head. I can figure it out but I have some other stuff to do first.

Comment (ljharb) [reply .github/workflows/esmeta-test262.yml:18]: `if: ` and then the condition; actually checking the condition might be tricky tho. doesn't have to be a blocker.

**Review by ljharb — APPROVED** [review]
skipping not a blocker

File: .github/workflows/esmeta-test262.yml (hunk 1 above)
Comment (ptomato) [reply .github/workflows/esmeta-test262.yml:18]: Found something that seems like it should work.

**Comment by ptomato** [comment]
Merging as per maintainers meeting

**Pull request merged**